tools/fileconv/api3/strategy.go

```diff
@@ -6,23 +6,87 @@ import (
 	"github.com/amp-labs/connectors/common/handy"
 )
 
-// This strategy prunes URL Path that should be omitted during Schema extraction.
-// It allows hard coded endpoint Path, as well as simple star rules focusing on matching prefix or suffix.
+type PathMatcher interface {
+	IsPathMatching(path string) bool
+}
+
+// PathMatchingStrategy This resolver has 2 level checking.
+// Path matches if it is approved by allow resolver or denied if matches by deny resolver.
+// The tie of what takes precedence is resolved by priorityAllow flag.
+// If neither match we rely on Strict flag.
+// Strict flag will mark a path as ignored, use this when AllowPaths and DenyPaths are exhaustive lists.
+// By default, Strict is false meaning any path will match, unless specified.
+//
+// prunes URL Path that should be omitted or included during Schema extraction.
+type PathMatchingStrategy struct {
```

Comment:
@RajatPawar Intercom was the first deep connector to require POST operations in OpenAPI. The solution with ignoreEndpoints doesn't hold that strongly. I was thinking to create some kind of imporvement and this PR was just to make Intercom use schema from POST.

This `PathMatchingStrategy` can be passed by either `ReadObjectsGet` or `ReadObjectsPost`.
It gives wider control on what endpoints the tool should look at and which to ignore. The strict flag would indicate if the end user of a method has provided exhaustive list or not. 
Usually you must explore all endpoints when creating metadata for ListObjectMetadata.
For intercom search endpoints are all that matters for POST extraction, so the list is strict, "exhaustive" and ok to ingore non-mentioned paths.
![image](https://github.com/user-attachments/assets/ec3a9383-e657-4b68-aae3-1e1a1f5eb20a)
![image](https://github.com/user-attachments/assets/7f08c75c-8240-4c99-86d6-792421c4f020)